File: packages/ai/src/tencent/ocr.ts
export interface TencentOcrOptions {
  readonly secretId: string;
  readonly secretKey: string;
  readonly timeoutMs?: number;
}

export interface OcrResult {
  readonly text: string;
  readonly items: readonly OcrItem[];
}

export interface OcrItem {
  readonly name: string;
  readonly value: string;
}

export async function recognizeReceipt(imageBase64: string, options: TencentOcrOptions): Promise<OcrResult> {
  const tencentcloud = await import("tencentcloud-sdk-nodejs");
  const OcrClient = tencentcloud.ocr.v20181119.Client;
  const client = new OcrClient({ credential: { secretId: options.secretId, secretKey: options.secretKey }, region: "ap-guangzhou" });
  const resp = await client.GeneralBasicOCR({ ImageBase64: imageBase64 });
  const items: OcrItem[] = (resp.TextDetections ?? []).map((det: any) => ({ name: "text", value: det.DetectedText ?? "" }));
  const text = items.map((i) => i.value).join("\n");
  return { text, items };
}

File: apps/api/ocr.py
import os

from tencentcloud.common import credential
from tencentcloud.ocr.v20181119 import ocr_client, models


def recognize_image(image_base64: str) -> str:
    """图片 base64 → 识别文字（腾讯云 OCR）"""
    cred = credential.Credential(
        os.environ["TENCENT_SECRET_ID"],
        os.environ["TENCENT_SECRET_KEY"],
    )
    client = ocr_client.OcrClient(cred, "ap-guangzhou")

    req = models.GeneralBasicOCRRequest()
    req.ImageBase64 = image_base64

    resp = client.GeneralBasicOCR(req)
    return "".join(item.DetectedText for item in resp.TextDetections)
